test/js/ui/map.test.js

```diff
@@ -292,6 +292,48 @@ test('Map', function(t) {
         }
     });
 
+    t.test('#setMinZoom', function (t) {
+        var map = createMap({zoom:0});
+        map.setMinZoom(3.5);
+        t.deepEqual(map.options.minZoom, 3.5);
+        t.end();
+
+        t.test('unset minZoom', function(t){
+            var map = createMap({minZoom:5});
+            map.setMinZoom()
+            t.deepEqual(map.options.minZoom, 0);
+            t.end();
+        })
+
+        t.test('ignore minZooms over maxZoom', function(t){
+            var map = createMap({maxZoom:5});
+            map.setMinZoom(6)
+            t.deepEqual(map.options.minZoom, 0);
+            t.end();
+        })
+    });
+
+    t.test('#setMaxZoom', function (t) {
+        var map = createMap({zoom:0});
+        map.setMaxZoom(3.5);
+        t.deepEqual(map.options.maxZoom, 3.5);
+        t.end();
+
+        t.test('unset maxZoom', function(t){
+            var map = createMap({maxZoom:5});
+            map.setMaxZoom()
+            t.deepEqual(map.options.maxZoom, 20);
+            t.end();
+        })
+
+        t.test('ignore maxZooms over minZoom', function(t){
+            var map = createMap({minZoom:5});
+            map.setMaxZoom(4)
+            t.deepEqual(map.options.minZoom, 20);
+            t.end();
+        })
+    });
```

Comment:
Best practice for unit tests is to refrain from using undocumented/private members, like `map.options`. (We don't always follow this best practice, unfortunately.) This gives us more flexibility to rejigger the `Map` class without breaking unit tests and makes our tests stronger as they are more representative of real-world usage.

Can you think of a way to rewrite these tests using only documented/public methods?